### ISSUE TITLE
DOES ANYONE HAVE A CACHED VERSION OF THE HANGOUTS EXTENSION STILL?

### ISSUE THREAD

**Issue by ajayyy** [old-hangouts/old-hangouts-chrome-extension#3]
If someone still has the cached version, we would be able to get the html/javascript to create it and that would make fixing the extension **much** easier.

Hope someone has this still. I got it back again, then lost it.

**Comment by ghost** [comment]
Is this what you're looking for?

https://www.dropbox.com/sh/wmg24g6enw51g5g/AACjIhlMnrsdcynQaTxRVw5-a?dl=0

**Comment by ajayyy** [comment]
@udpPigeons No, if your hangouts still does not display "This extension is old" then somehow find your chrome cache because that would have the files that make up the extension that have been deleted from Google.

**Comment by ghost** [comment]
Okay. If I come across it, I'll let you know!

**Comment by pranav-prakash** [comment]
Wait, since the old version was essentially just an iframe to a website generated dynamically by google, what would the benefit of having access to this be?

I don't think it's possible to access the old ui anymore. Perhaps the only option is to create our own hybrid extension fitting the new UI in the panel interface.

I tried going to the iFrame directly and modifying the URL a bit this is the only thing I could salvage:

`
<!DOCTYPE html><html><head><meta http-equiv="X-UA-Compatible" content="IE=Edge"/></head><body onload="init()"><script type="text/javascript">function init() {var frameType = window.location.hash[1]; var callbackWindow; switch (frameType) {case 't': callbackWindow = window.parent.opener; break; case 'p': callbackWindow = window.open('', 'gtn-roster-iframe-id'); break; case 'e': callbackWindow = window.parent.frames['gtn-roster-iframe-id']; break; case 'n': callbackWindow = null; break; default: throw Error('Unknown frame type: ' + frameType);}if (callbackWindow != null) {callbackWindow['_GC_OnFrameReady'](window);}}</script></body></html>`

**Comment by ajayyy** [comment]
@pranav-prakash This is exactly what I was trying to do earlier as noted in #2. I tried to replace the iframe with the new extension and the gmail/hangouts.google.com version but it gave errors because they included new functions in the javascript and it just displayed blank. I might try more tomorrow or some other time. I think **this** is the best solution we are going to have. And it should not be hard to do.

**Comment by ghost** [comment]
I suspect there was something missing in your experiment. Perhaps attempting to include all external JS and CSS references might help, seeing as extra links to such items wouldn't hurt it.

**Comment by ajayyy** [comment]
@pranav-prakash One of the javascript files would need to be merged. I didn't have enough time to do that. I might try it later though.

**Comment by ajayyy** [comment]
@udpPigeons One of the javascript files would need to be merged. I didn't have enough time to do that. I might try it later though.

**Comment by pranav-prakash** [comment]
@ajayyy 

Got it (sort of?) working.

https://github.com/pranav-prakash/Old-Hangouts

Basically in datachannel.js I changed the iframe url parameter from "aChromeExtension" to "ChromeApp" after observing the difference in requests between the two clients.

It now works and displays contact list + messages but actually clicking on those to open a chat window doesn't work. I think the js returned from the server needs to be modified to open it in a new panel. The chrome dev tool console doesn't show any warnings/errors though.

Also I don't know how long this method will last.

**Comment by ajayyy** [comment]
@pranav-prakash That's really cool. That's weird that you have to change the iframe url parameter.

**Comment by pranav-prakash** [comment]
@ajayyy I'm not sure how to get it to open a new panel when you click on a contact though. Reverse engineering it is pretty hard.

**Comment by sandoche** [comment]
@ajayyy I tried your repo it's very cool, I'll have a look
